ios/CodePush/CodePush.m

```diff
@@ -455,9 +455,12 @@ - (void)rollbackPackage
 {
     NSError *error;
     NSDictionary *failedPackage = [CodePushPackage getCurrentPackage:&error];
-
-    // Write the current package's metadata to the "failed list"
-    [self saveFailedUpdate:failedPackage];
+    if (error) {
```

Comment:
It's a code smell to check the error pointer directly, and yes, it does happen occasionally that a method will return a non-nil error object even if the call succeeds. https://www.bignerdranch.com/blog/an-nserror-error/

An alternative here would be this:

```
NSError *error;
NSDictionary *failedPackage = [CodePushPackage getCurrentPackage:&error];

if (!failedPackage) {
    CPLog(@"Error getting current update metadata during rollback: %@", error);
} else {
    // Write the current package's metadata to the "failed list"
    [self saveFailedUpdate:failedPackage];
}
```